libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -80,21 +106,56 @@ interface SelectableRow {
 @Component({
     selector: 'fdp-table',
     templateUrl: './table.component.html',
-    changeDetection: ChangeDetectionStrategy.OnPush
+    styleUrls: ['./table.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [TableService],
+    host: {
+        '[class.fd-table]': 'true',
+        '[class.fd-table--compact]': 'contentDensity === CONTENT_DENSITY.COMPACT',
+        '[class.fd-table--condensed]': 'contentDensity === CONTENT_DENSITY.CONDENSED',
+        '[class.fd-table--no-horizontal-borders]': 'noHorizontalBorders || noBorders',
+        '[class.fd-table--no-vertical-borders]': 'noVerticalBorders || noBorders'
+    }
 })
-export class TableComponent implements OnChanges, OnInit, AfterViewInit, OnDestroy, CssClassBuilder {
+export class TableComponent<T = any> implements AfterViewInit, OnDestroy, OnChanges {
```

Comment:
Do you see any usecase, where Table will use other type than `any`  99% of usecase will be using `fdp-xxx` in html template, and the rest programatic - not sure.

Reply:
This `any` is needed to type `dataSource` properly because dataSource is a generic thing.
We have this type `T` to offer a user referring to the table instance in a safe way. By default it is an `any` so it does not push a user to define a type.
So if user strives to be accurate in his code he migt write the next:
```
@ViewChild(TableComponent)
table: TableComponent<DataItemType>;
```